### PR TITLE
MNT: Add CI mode to examples

### PR DESCRIPTION
**Pull request by mtsokol**
This PR adds CI mode to examples reducing `examples` job execution time (each benchmark returns after precompilation stage).

### CODE REVIEW

File: examples/utils.py [hunk 1]
```diff
@@ -1,9 +1,16 @@
+import os
 import time
 from collections.abc import Callable, Iterable
 from typing import Any
 
+CI_MODE = os.getenv("CI_MODE", default=False)
```
Comment (hameerabbasi) [inline examples/utils.py:6]:
```suggestion
CI_MODE = bool(int(os.getenv("CI_MODE", default="0")))
```

File: ci/test_examples.sh [hunk 2]
```diff
@@ -1,3 +1,3 @@
 for example in $(find ./examples/ -iname *.py); do
-  python $example
+  CI_MODE=True python $example
```
Comment (hameerabbasi) [inline ci/test_examples.sh:2]:
```suggestion
  CI_MODE=1 python $example
```

File: examples/utils.py [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
+
 
 def benchmark(func: Callable, args: Iterable[Any], info: str, iters: int):
+    if CI_MODE:
```
Comment (hameerabbasi) [inline examples/utils.py:10]: We should have a warmup here.

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
Thanks for the PR, @mtsokol.

File: examples/utils.py (hunk 3 above)
Comment (mtsokol) [reply examples/utils.py:10]: Warmup, or a precompilation stage, happens in each example file before `benchmark` is called. It isn't placed here as we also want the result of that call for `assert_allclose` statements.

File: ci/test_examples.sh (hunk 2 above)
Comment (mtsokol) [reply ci/test_examples.sh:2]: Is there a difference compared to the notebook job that we merged?

https://github.com/pydata/sparse/blob/0e95f8b5c6e98f73bfb2c5faba71c4f90abc17af/ci/test_notebooks.sh#L1

We use `True` there. Or do we want to also change it here?

Comment (hameerabbasi) [reply ci/test_examples.sh:2]: I'd say also change it there. If there's repetition we should factor it out.

File: examples/utils.py (hunk 3 above)
Comment (hameerabbasi) [reply examples/utils.py:10]: We could return from this function which is the result of the warmup call.

Comment (mtsokol) [reply examples/utils.py:18]: Done!

File: ci/test_examples.sh (hunk 2 above)
Comment (mtsokol) [reply ci/test_examples.sh]: Done!

File: examples/utils.py (hunk 1 above)
Comment (mtsokol) [reply examples/utils.py]: Done!

**Review by hameerabbasi — APPROVED** [review]
Thanks for the work here, @mtsokol!

**Pull request merged**